wp-includes/parser/class-parse-tree.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+class Parse_Tree {
+	public $rule_id;
+	public $rule_name;
+	public $children = array();
+
+	public function __construct( $rule_id, $rule_name ) {
+		$this->rule_id   = $rule_id;
+		$this->rule_name = $rule_name;
+	}
+
+	public function append_child( $node ) {
+		$this->children[] = $node;
+	}
+
+	/**
+	 * Flatten the matched rule fragments as if their children were direct
+	 * descendants of the current rule.
+	 *
+	 * What are rule fragments?
+	 *
+	 * When we initially parse the BNF grammar file, it has compound rules such
```

Comment:
Note we're no longer using the BNF format and this comment could use some brushing up